src/raftstore/store/worker/region.rs

```diff
@@ -103,6 +104,38 @@ impl SnapContext {
         let raft_db = Arc::clone(&self.raft_db);
         let raw_snap = Snapshot::new(Arc::clone(&self.kv_db));
 
+        let mut old_snaps = None;
+        while self.mgr.get_total_snap_size() > self.mgr.max_total_snap_size() {
+            if old_snaps.is_none() {
+                let snaps = match self.mgr.list_idle_snap() {
+                    Ok(snaps) => snaps,
+                    Err(_) => break,
+                };
+                let mut key_and_snaps = Vec::with_capacity(snaps.len());
+                for (key, is_sending) in snaps {
+                    if !is_sending {
+                        continue;
+                    }
+                    let snap = match self.mgr.get_snapshot_for_sending(&key) {
+                        Ok(snap) => snap,
+                        Err(_) => continue,
+                    };
+                    if !snap.meta().and_then(|m| m.modified()).is_ok() {
```

Comment:
Better save the modified time.

Reply:
I think save `Snapshot` is better because then we can call `self.delete_snapshot(&snap)` directly.